docs/connectors/io-crd-config/sink-crd-config.md

```diff
@@ -63,30 +63,44 @@ If the node where a Pod is running has enough of a resource available, it is pos
 
 ## Secrets
 
-In Function Mesh, the secret is defined through a secretsMap. To use a secret, a Pod needs to reference the secret. Pods can consume secretsMaps as environment variables in a volume. You can specify the `data` field when creating a configuration file for a secret.
+Function Mesh provides the `SecretsMap` field for Function, Source and Sink in the CRD definition. You can refer to the created secret under the same namespace and the controller can include those referred secrets as environment variables. You can specify the `data.username` and `data.password` fields when creating a secret, as shown below.
 
-To use a secret in an environment variable in a Pod, follow these steps.
+```yaml
+apiVersion: v1
+data:
+  username: <secret_key>
+  password: <secret_password>
+kind: Secret
+metadata:
+  name: <secret_name>
+type: Opaque
+```
 
-1. Create a secret or use an existing one. Multiple Pods can reference the same secret.
-2. Modify your Pod definition in each container, which you want to consume the value of a secret key, to add an environment variable for each secret key that you want to consume.
-3. Modify your image and or command line so that the program looks for values in the specified environment variables.
+This table lists configurations about the `SecretsMap` field.
```

Comment:
`SecretsMap` is defined as a `Map` struct with `String` keys and `SecretReference` values. The `key` will indicate the environment value in the container, and the `SecretReference` is defined as below.

| Field | Description |
 | --- | --- |
 | `path` | The name of the secret in the pod's namespace to select from. |
 | `key` | The key of the secret to select from.  Must be a valid secret key. |

For example, the user has a Kubernetes Secret named `credential-secret` defined as below:

```yaml
 apiVersion: v1
 data:
   username: foo
   password: bar
 kind: Secret
 metadata:
   name: credential-secret
 type: Opaque
 ```

To use it in Pulsar Functions in a security way, simply define the `SecretsMap` in the Custom Resource:

```
secretsMap:
  username:
    path: credential-secret
    key: username
  password:
    path: credential-secret
    key: password
```

Then in the Pulsar Functions and Connectors, you can call `context.getSecret("username")` to get the secret value.